src/components/Post/Post.js

```diff
@@ -1,115 +1,106 @@
+    state = {
+        comments: [],
+        isCommentsLoading: false,
+        commentsLoaded: false,
+        commentsSectionExpanded: false,
+        error: ""
+    };
 
-  onLoadComments = () => {
-    // todo достать из props пропсу item, использовать item.id в запросе ниже в строке 39
+    onLoadComments = () => {
+        const {item} = this.props;
 
-    //todo поменять стейт так, чтоб было понятно что секция с комментариями открыта и началась загрузка
-    //todo т.е. isCommentsLoading и commentsSectionExpanded станут true
+        this.setState({
+            isCommentsLoading: true,
+            commentsSectionExpanded: true
+        });
 
-    fetch(`https://gorest.co.in/public-api/comments?access-token=${accessToken}&post_id=${1/* todo тут вместо 1 будет id поста т.е. item.id */}`, )
-      .then(res => {
-        if (!res.ok) throw Error(res.statusText);
-        return res.json();
-      })
-      .then(data => {
-        // todo поменять стейт так, чтоб в comments лежали data.result,
-        // todo лоадинг закончился, т.е. isCommentsLoading будет false, а commentsLoaded станет true (т.е. запрос был выполнен)
-      })
-      .catch(error => {
-        console.log(error);
-        // todo поменять стейт так, чтоб
-        // todo лоадинг закончился, т.е. isCommentsLoading будет false, а commentsLoaded станет false (т.е. запрос не был выполнен ввиду ошибки)
-        // todo в error пойдет значение, error которая вывалилась в результате запроса
-        // todo и закроем секцию коментариев т.е. commentsSectionExpanded будет false
-      });
-  };
+        fetch(`https://gorest.co.in/public-api/comments?access-token=${accessToken}&post_id=${item.id}`,)
+            .then(res => {
+                if (!res.ok) throw Error(res.statusText);
+                return res.json();
+            })
+            .then(data => {
+                this.setState({
+                    comments: data.result,
+                    isCommentsLoading: false,
+                    commentsLoaded: true
+                })
+            })
+            .catch(error => {
+                console.log(error);
+                this.setState({
+                    isCommentsLoading: false,
+                    commentsLoaded: false,
+                    commentsSectionExpanded: false
+                })
+            });
+    };
 
   onShowComments = () => {
-    // todo если запрос был выполнен раннее, т.е. commentsLoaded = true
-    //    меняем в стейт значение commentsSectionExpanded на противоположное
-    //    делаем return; чтоб остально код этой функции не выполнялся
-    //    т.е. тоглим отображение комментариев
-
-    // todo иначе, если запроса не было выполняем его, вызвав метод onLoadComments
+      const {commentsLoaded, commentsSectionExpanded} = this.state;
+      if (!commentsLoaded) {
+          return this.onLoadComments()
+      } else {
+          return this.setState({commentsSectionExpanded: !commentsSectionExpanded});
```

Comment:
You can skip `return` here just use 
`this.onLoadComments()`
and 
`this.setState({commentsSectionExpanded: !commentsSectionExpanded});`